static/app/utils/useApiRequests.tsx

```diff
@@ -305,69 +209,125 @@ function useApiRequests({
+            query = {...locationQuery, ...query};
+          }
+          try {
+            const results = await api.requestPromise(endpoint, {
+              method: 'GET',
+              ...parameters,
+              query,
+              includeAllArgs: true,
+            });
+            const [data, _, resp] = results;
+            handleRequestSuccess({stateKey, data, resp}, true);
+          } catch (error) {
+            handleError(error, [stateKey, endpoint, parameters, options]);
+          }
+        })
+      );
+    },
+    [api, endpoints, handleError, handleRequestSuccess, location]
+  );
+
+  const reloadData = useCallback(() => fetchData({isReloading: true}), [fetchData]);
+
+  const handleMount = useCallback(async () => {
+    try {
+      await fetchData();
+    } catch (error) {
+      setState(prevState => ({...prevState, hasError: true}));
+      throw error;
     }
+  }, [fetchData]);
 
-    // Cancel any in flight requests
-    api.clear();
+  // Trigger fetch on mount
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => void handleMount(), []);
 
-    setState(prevState => ({
-      ...prevState,
-      isLoading: true,
-      hasError: false,
-      remainingRequests: endpoints.length,
-      ...extraState,
-    }));
-
-    await Promise.all(
-      endpoints.map(async ([stateKey, endpoint, parameters, options]) => {
-        options = options || {};
-        // If you're using nested async components/views make sure to pass the
-        // props through so that the child component has access to props.location
-        const locationQuery = (location && location.query) || {};
-        let query = (parameters && parameters.query) || {};
-        // If paginate option then pass entire `query` object to API call
-        // It should only be expecting `query.cursor` for pagination
-        if ((options.paginate || locationQuery.cursor) && !options.disableEntireQuery) {
-          query = {...locationQuery, ...query};
-        }
-        try {
-          const results = await api.requestPromise(endpoint, {
-            method: 'GET',
-            ...parameters,
-            query,
-            includeAllArgs: true,
-          });
-          const [data, _, resp] = results;
-          handleRequestSuccess({stateKey, data, resp}, true);
-        } catch (error) {
-          handleError(error, [stateKey, endpoint, parameters, options]);
-        }
-      })
-    );
-  }
+  const handleFullReload = useCallback(() => {
+    if (shouldReload) {
+      return reloadData();
+    }
+    setState({...initialState});
+    return fetchData();
+  }, [initialState, reloadData, fetchData, shouldReload]);
 
-  const shouldRenderLoading = state.isLoading && (!shouldReload || !state.isReloading);
+  // Trigger fetch on location or parameter change
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => void handleFullReload(), [location.search, location.state, params]);
```

Comment:
```suggestion
 // Trigger fetch on location or parameter change
  useEffect(
    () => void handleFullReload(),
    [location.search, location.state, params, handleFullReload]
  );

```

Reply:
I'll follow up with some comments as to why these are disabled